tools/clean.mjs

```diff
@@ -8,11 +8,48 @@
 
 import {exec} from 'child_process';
 import {readdirSync} from 'fs';
+import readline from 'node:readline';
 
-exec(
-  `git clean -Xf ${readdirSync(process.cwd())
-    .filter(file => {
-      return file !== 'node_modules';
-    })
-    .join(' ')}`
-);
+let resolver;
+let rejector;
+const confirmation = new Promise((res, rej) => {
+  resolver = res;
+  rejector = rej;
+});
+
+if (process.env.npm_command !== 'run-script') {
```

Comment:
why is it important if it is run via npm or not? let's just check if `git rev-parse --show-toplevel` is the process.cwd() and abort cleaning if git root is wrong

Reply:
It still can be invoked without .git folder in puppeteer with an npm script defined elsewhere but the process.cwd() cannot probably used for checking since it could be a subfolder in Puppeteer

Reply:
let's check if the directory in which tools is located is the same as `git rev-parse --show-toplevel` and bail otherwise

Reply:
maybe we should list targets for deletion explicitly but it would be somewhat tedious to maintain it

Reply:
Done.